clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -1030,6 +1033,76 @@ test_regions_core_sharded()
     assert(ordinal == 5);
 }
 
+static void
+test_regions_by_shard()
+{
+    std::cerr << "\n----------------\nTesting ROI specified by shard\n";
+    static constexpr int NUM_WORKLOADS = 2;
+    static constexpr int NUM_CORES_PER_WORKLOAD = 2;
+    static constexpr int NUM_OUTPUTS = NUM_WORKLOADS * NUM_CORES_PER_WORKLOAD;
+    static constexpr int NUM_ORIGINAL_INPUTS = 3;
+    static constexpr int NUM_INSTRS = 9;
+    static constexpr memref_tid_t TID_BASE = 100;
+    std::vector<scheduler_t::input_workload_t> sched_inputs;
+    // This is core-sharded with interleaved threads on each core.
+    for (int workload_idx = 0; workload_idx < NUM_WORKLOADS; workload_idx++) {
+        std::vector<scheduler_t::input_reader_t> readers;
+        for (int core_idx = 0; core_idx < NUM_CORES_PER_WORKLOAD; core_idx++) {
+            std::vector<trace_entry_t> inputs;
+            for (int input_idx = 0; input_idx < NUM_ORIGINAL_INPUTS; input_idx++) {
+                inputs.push_back(make_thread(TID_BASE + input_idx));
+                inputs.push_back(make_pid(1));
+            }
+            for (int instr_idx = 0; instr_idx < NUM_INSTRS; instr_idx++) {
+                for (int input_idx = 0; input_idx < NUM_ORIGINAL_INPUTS; input_idx++) {
+                    inputs.push_back(make_thread(TID_BASE + input_idx));
+                    inputs.push_back(make_instr(42 + instr_idx * 4));
+                }
+            }
+            for (int input_idx = 0; input_idx < NUM_ORIGINAL_INPUTS; input_idx++) {
+                inputs.push_back(make_exit(TID_BASE + input_idx));
+            }
+            readers.emplace_back(
+                std::unique_ptr<mock_reader_t>(new mock_reader_t(inputs)),
+                std::unique_ptr<mock_reader_t>(new mock_reader_t()), -1 /*sentinel*/);
+        }
+        sched_inputs.emplace_back(std::move(readers));
+    }
+    // Set up different skips on each input, increasing by one as we go.
+    for (int workload_idx = 0; workload_idx < NUM_WORKLOADS; workload_idx++) {
+        std::vector<scheduler_t::input_reader_t> readers;
+        for (int core_idx = 0; core_idx < NUM_CORES_PER_WORKLOAD; core_idx++) {
+            std::vector<scheduler_t::range_t> regions;
+            regions.emplace_back(
+                1 /*1-based*/ + workload_idx * NUM_CORES_PER_WORKLOAD + core_idx, 0);
+            scheduler_t::input_thread_info_t mod(regions);
+            mod.shards = std::vector<int>(1, core_idx);
```

Comment:
nit: ` = {core_idx}` would be simpler.

Reply:
Done.  Somehow that didn't work for the tids in another place; but it works here.